docs/book/src/developer/providers/contracts/infra-cluster.md

```diff
@@ -0,0 +1,516 @@
+- `attributes map[string]string`: arbitrary attributes for users to apply to a failure domain.
+
+Once `spec.failureDomains` is set on the InfraCluster resource and the [InfraCluster initialization completed],
+the Cluster controller will bubble up this info in Cluster's `status.failureDomains`.
+
+### InfraCluster initialization completed
+
+Each InfraCluster MUST report when Cluster's infrastructure is fully provisioned (initialization) by setting
+`status.ready` in the InfraCluster resource.
+
+```go
+type FooClusterStatus struct {
+    // Ready denotes that the foo cluster infrastructure fully provisioned.
+    // +optional
+    Ready bool `json:"ready"`
+    
+    // See other rules for more details about mandatory/optional fields in InfraCluster status.
+    // Other fields SHOULD be added based on the needs of your provider.
+}
+```
+
+Once `status.ready` the Cluster "core" controller will bubbles up this info in Cluster's `status.infrastructureReady`;
+If defined, also InfraCluster's `spec.controlPlaneEndpoint` and `spec.failureDomains` will be surfaced on Cluster's
+corresponding field at the same time.
+
+<aside class="note warning">
+
+<h1>Heads up! this will change with the v1beta2 contract</h1>
+
+When the v1beta2 contract will be released (tentative Apr 2025), `status.initialization.provisioned` will be used
+instead of `status.ready`. However, `status.ready` will be supported until v1beta1 removal (~one year later).
+
+See [Improving status in CAPI resources].
+
+</aside>
+
+### InfraCluster conditions
+
+According to [Kubernetes API Conventions], Conditions provide a standard mechanism for higher-level
+status reporting from a controller.
+
+Providers implementers SHOULD implement `status.conditions` for their InfraCluster resource.
+In case conditions are implemented, Cluster API condition type MUST be used.
+
+If a condition with type `Ready` exist, such condition will be mirrored in Cluster's `InfrastructureReady` condition.
+
+Please note that the `Ready` condition is expected to surface the status of the InfraCluster during its own entire lifecycle,
+including initial provisioning, the final deletion process, and the period in between this two moments.
+
+See [Cluster API condition proposal] for more context.
+
+<aside class="note warning">
+
+<h1>Heads up! this will change with the v1beta2 contract</h1>
+
+When the v1beta2 contract will be released (tentative Apr 2025), Cluster API will start using Kubernetes metav1.Condition
+types and fully comply to [Kubernetes API Conventions].
+
+In order to support providers continuing to use legacy Cluster API condition types, providers transitioning to
+metav1.Condition or even providers adopting custom condition types, Cluster API will start to accept `Ready` condition that
+provides following information:
+- `type`
+- `status`
+- `reason` (if omitted, a default one will be used)
+- `message` (optional)
+
+Other fields will be ignored
```

Comment:
Is this true? What about LastTransitionTime?

```
			// NOTE: we are preserving the original transition time (when the underlying condition changed)
			LastTransitionTime: condition.LastTransitionTime,
```

Reply:
Fixed